master/buildbot/data/types.py

```diff
@@ -156,7 +156,7 @@ def valueFromString(self, arg):
 class Identifier(Type):
 
     name = "identifier"
-    identRe = re.compile('^[a-zA-Z_-][a-zA-Z0-9_-]*$')
+    identRe = re.compile('^[a-zA-Z._-][a-zA-Z0-9._-]*$')
```

Comment:
start with '.' looks weird for a buildername

Reply:
I initially included this one, but I could not figure out where it was used. Thanks for re-discovering it :)
(Maybe it would help to at least comment on where this was used, or is it too broad?)

Isn't `__Janitor` as a builder name, or a builder name starting with a dash (`-`) equally weird as a builder name starting with a dot? I do imagine that some users might for whatever reason use a leading dot (maybe also to distinguish it from other regular builders), and unless there are technical difficulties with that, I don't see a good reason to prevent it (just because it looks weird).

Reply:
> Isn't __Janitor as a builder name, or a builder name starting with a dash (-) equally weird as a builder name starting with a dot?
True.

It is called identifier, and it is starting to become less and less an identifier.

At this point, we could also just change all those identifier types to unrestricted strings, as I am not sure there is really any technical issue beyond that

Reply:
Hmm, starting with '.' does look weird. Variable names are usually allowed to start with letters or '_'. Same concept can go for buildernames in data api query. I had one doubt: how are we able to use dots in buildernames although it isn't present in the identifiers at https://github.com/buildbot/buildbot/blob/master/master/buildbot/util/identifiers.py#L20 ?